2011-11-02  Andrew Stubbs  <ams@codesourcery.com>

	Backport from FSF mainline:

	2011-11-01  Andrew Stubbs  <ams@codesourcery.com>

	gcc/
	* config/arm/bpabi.h (BE8_LINK_SPEC): Recognize generic-armv7 tuning.

=== modified file 'gcc/config/arm/bpabi.h'
--- old/gcc/config/arm/bpabi.h	2010-12-20 17:48:51 +0000
+++ new/gcc/config/arm/bpabi.h	2011-11-02 21:02:53 +0000
@@ -56,7 +56,8 @@
   "|march=armv4|mcpu=fa526|mcpu=fa626:--fix-v4bx}"
 
 #define BE8_LINK_SPEC " %{mbig-endian:%{march=armv7-a|mcpu=cortex-a5"\
-  "|mcpu=cortex-a8|mcpu=cortex-a9|mcpu=cortex-a15:%{!r:--be8}}}"
+  "|mcpu=cortex-a8|mcpu=cortex-a9|mcpu=cortex-a15|mcpu=generic-armv7-a"\
+  ":%{!r:--be8}}}"
 
 /* Tell the assembler to build BPABI binaries.  */
 #undef  SUBTARGET_EXTRA_ASM_SPEC

